src/common/pf_snmp.c

```diff
@@ -134,22 +134,80 @@ static void pf_snmp_encode_signal_delays (
 
 void pf_snmp_get_system_name (pnet_t * net, pf_snmp_system_name_t * name)
 {
+   const char * directory = net->fspm_cfg.file_directory;
    int error;
 
    CC_STATIC_ASSERT (sizeof (name->string) >= PNAL_HOSTNAME_MAX_SIZE);
 
-   error = pnal_get_hostname (name->string);
+   error = pf_file_load (directory, PF_FILENAME_SYSNAME, name, sizeof (*name));
    if (error)
    {
-      name->string[0] = '\0';
+      error = pnal_get_hostname (name->string);
+      if (error)
+      {
+         name->string[0] = '\0';
+      }
+      LOG_INFO (
+         PF_SNMP_LOG,
+         "SNMP(%d): Could not yet read sysName from nvm.\n",
+         __LINE__);
+   }
+   else
+   {
+      LOG_INFO (PF_SNMP_LOG, "SNMP(%d): Did read sysName from nvm.\n", __LINE__);
    }
    name->string[sizeof (name->string) - 1] = '\0';
 }
 
 int pf_snmp_set_system_name (pnet_t * net, const pf_snmp_system_name_t * name)
 {
-   /* TODO: Set new hostname permanently. Perhaps call pnal_set_ip_suite()? */
-   return -1;
+   const char * directory = net->fspm_cfg.file_directory;
+   pf_snmp_system_name_t temporary_buffer;
+   int res;
+
```

Comment:
`/* TODO: Possibly also set hostname via pnal_set_ip_suite() */`

Reply:
Setting hostname would probably not be correct as sysName (by convention) should be the fully qualified domain name, not the hostname.
Added comment instead.